pkg/overlay/cache.go

```diff
@@ -148,3 +149,23 @@ func (cache *Cache) Delete(ctx context.Context, id storj.NodeID) error {
 
 	return nil
 }
+
+// ConnFailure implements the Transport Observer `ConnFailure` function
+func (cache *Cache) ConnFailure(ctx context.Context, node *pb.Node, err error) {
+	// TODO: noop until we figure out what to do on ConnFailure
+	// Kademlia paper specifies 5 unsuccessful PINGs before removing the node
+	// from our routing table, but this is the cache so maybe we want to treat
+	// it differently.
```

Comment:
We need to update uptime in statdb when we fail to connect to a node. Same as the other one, but with `false` instead of `true`

```
_, err := cache.statdb.UpdateUptime(ctx, node.Id, false)
if err != nil {
	// log statdb error
}
```

Reply:
Added.